src/coreclr/src/binder/inc/applicationcontext.hpp

```diff
@@ -46,7 +46,11 @@ namespace BINDER_SPACE
             key = e.m_wszSimpleName;
             return key;
         }
-        static count_t Hash(const key_t &str) { return HashiString(str); }
+        static count_t Hash(const key_t &str)
+        {
+            SString ssKey(str);
```

Comment:
```suggestion
            SString ssKey(SString::Literal, str);
```
This may fix the assert you are seeing.